### PR TITLE
Update HTTP Publishing section

### PR DESCRIPTION
**Pull request by jugmac00**
Zope 4 no longer ships with the ZServer by default.

Instead, Zope interacts with a WSGI middleware.

The description and the figure has been updated accordingly.

modified:   docs/zdgbook/Figures/2-1.png
modified:   docs/zdgbook/ObjectPublishing.rst

### CODE REVIEW

File: docs/zdgbook/ObjectPublishing.rst [hunk 1]
```diff
@@ -34,21 +34,31 @@ web publishing.
 HTTP Publishing
 ===============
 
-When you contact Zope with a web browser, your browser sends an HTTP
-request to Zope's web server. After the request is completely
-received, it is processed by 'ZPublisher', which is Zope's object
-publisher. 'ZPublisher' is a kind of light-weight ORB (Object Request
+Zope 4 no longer ships with a builtin web server, so when you want
+to interact with Zope via browser you have to setup a WSGI server as
+a middleware.
```
Comment (icemac) [inline docs/zdgbook/ObjectPublishing.rst]: I don't think that the server is a middleware. See [this graphical explanation](https://image.slidesharecdn.com/wsgi-150310012800-conversion-gate01/95/python-wsgi-introduction-3-1024.jpg?cb=1479297785 ) I found. (This image was taken from https://www.slideshare.net/AgeeleshwarK/python-wsgi)

Comment (dataflake) [reply docs/zdgbook/ObjectPublishing.rst]: Just leave out the last part of that sentence, "as a middleware". My understanding is that "middleware" is software in the WSGI stack that the request passes through on the way to or from Zope.

Comment (icemac) [reply docs/zdgbook/ObjectPublishing.rst]: That's my understanding, too.

Comment (jugmac00) [reply docs/zdgbook/ObjectPublishing.rst]: @icemac and @dataflake - thank you for the time to review the PR and offer some improvements.

Obviously, I misunderstood the term "middleware" in this context - my thinking was that a middleware (alias for WSGI server e.g. gunicorn) is the part which manages the communication between the web server (eg. Nginx) and the application (eg. Zope/Flask).

After reading a couple of blog posts and the PEP 3333 I get a better picture.

Gonna update the text in a sec.

Thanks again for your time. I hope updating the documentation is a win-win for all of us - I learn a lot about the Zope stack and web development, and from the Zope side... the documentation gets updated for the first time in years :-)

File: docs/zdgbook/ObjectPublishing.rst [hunk 2]
```diff
@@ -34,21 +34,30 @@ web publishing.
 HTTP Publishing
 ===============
 
-When you contact Zope with a web browser, your browser sends an HTTP
-request to Zope's web server. After the request is completely
-received, it is processed by 'ZPublisher', which is Zope's object
-publisher. 'ZPublisher' is a kind of light-weight ORB (Object Request
+Zope 4 no longer ships with a builtin web server, so when you want
+to interact with Zope via browser you have to setup a WSGI server as.
```
Comment (ale-rt) [inline docs/zdgbook/ObjectPublishing.rst]: it seems the trailing "as" should be removed.

Comment (jugmac00) [reply docs/zdgbook/ObjectPublishing.rst]: Thank you @ale-rt

**Comment by jugmac00** [comment]
@icemac and @dataflake - changes applied.

**Pull request merged**